File: Extended-TS-SDK-master/src/utils/date.ts
/**
 * Date and time utility functions
 */

/**
 * Get current UTC datetime
 */
export function utcNow(): Date {
  return new Date();
}

/**
 * Convert Date to epoch milliseconds
 * Uses Math.ceil to ensure consistent rounding behavior
 */
export function toEpochMillis(value: Date): number {
  return Math.ceil(value.getTime());
}

File: libs/python_sdk-starknet/x10/utils/date.py
import math
from datetime import UTC, datetime


def utc_now():
    return datetime.now(tz=UTC)


def to_epoch_millis(value: datetime):
    assert value.tzinfo == UTC, "`value` must be in UTC"

    # Use ceiling to match the hash_order logic which uses math.ceil
    return int(math.ceil(value.timestamp() * 1000))
